### PR TITLE
Add feature: Start group at runtime

### PR DESCRIPTION
**Pull request by JimmyShi22**

### CODE REVIEW

File: fisco-bcos/main/main.cpp [hunk 1]
```diff
@@ -105,6 +106,16 @@ string initCommandLine(int argc, const char* argv[])
     return configPath;
 }
 
+void checkAndCall(const std::string& configPath, shared_ptr<Initializer> initializer)
+{
+    (void)initializer;
```
Comment (bxq2011hust) [inline fisco-bcos/main/main.cpp]:
```suggestion

```

Comment (JimmyShi22) [reply fisco-bcos/main/main.cpp]: deleted

File: libinitializer/LedgerInitializer.cpp [hunk 2]
```diff
@@ -92,19 +95,62 @@ void LedgerInitializer::initConfig(boost::property_tree::ptree const& _pt)
...
+        }
+    }
+    return reachList;
+}
+
+void LedgerInitializer::startMoreLedger()
+{
+    vector<dev::GROUP_ID> newGroupIDList;
+    assert(m_groupConfigPath.size() != 0);
+    newGroupIDList = initLedgers();
```
Comment (cyjseagull) [inline libinitializer/LedgerInitializer.cpp]: recommend `vector<dev::GROUP_ID> newGroupIDList = initLedgers();`

File: libinitializer/LedgerInitializer.cpp [hunk 3]
```diff
@@ -32,56 +32,59 @@ using namespace dev::initializer;
...
 void LedgerInitializer::initConfig(boost::property_tree::ptree const& _pt)
 {
-    namespace fs = boost::filesystem;
     INITIALIZER_LOG(DEBUG) << LOG_BADGE("LedgerInitializer") << LOG_DESC("initConfig");
-    m_groupDataDir = _pt.get<string>("group.group_data_path", "data/");
-    auto groupConfigPath = _pt.get<string>("group.group_config_path", "conf/");
+
     assert(m_p2pService);
+    m_groupDataDir = _pt.get<string>("group.group_data_path", "data/");
+    m_groupConfigPath = _pt.get<string>("group.group_config_path", "conf/");
```
Comment (cyjseagull) [inline libinitializer/LedgerInitializer.cpp]: I think `assert(m_groupConfigPath.size() != 0)` should be here.

File: libinitializer/LedgerInitializer.cpp [hunk 4]
```diff
@@ -32,56 +32,59 @@ using namespace dev::initializer;
...
     {
-        LOG(INFO) << LOG_BADGE("LedgerInitializer") << LOG_KV("groupConfigPath", groupConfigPath);
-        fs::path path(groupConfigPath);
-        if (fs::is_directory(path))
-        {
-            fs::directory_iterator endIter;
-            for (fs::directory_iterator iter(path); iter != endIter; iter++)
-            {
-                if (fs::extension(*iter) == ".genesis")
+        newGroupIDList = foreachLedgerConfigure(
```
Comment (cyjseagull) [inline libinitializer/LedgerInitializer.cpp]: Why not `vector<dev::GROUP_ID> newGroupIDList = foreachLedgerConfigure`

File: fisco-bcos/main/FileSignal.h [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
```
[the remark this answers is not in the document]
Comment (JimmyShi22) [reply fisco-bcos/main/FileSignal.h]: done

File: libinitializer/LedgerInitializer.cpp (hunk 3 above)
Comment (JimmyShi22) [reply libinitializer/LedgerInitializer.cpp]: done

File: libinitializer/LedgerInitializer.cpp (hunk 2 above)
Comment (JimmyShi22) [reply libinitializer/LedgerInitializer.cpp]: done

File: fisco-bcos/main/main.cpp [hunk 6]
```diff
@@ -105,6 +106,15 @@ string initCommandLine(int argc, const char* argv[])
     return configPath;
 }
 
+void checkAndCall(const std::string& configPath, shared_ptr<Initializer> initializer)
+{
+    std::string moreGroupSignal = configPath + ".start_more_group";
```
Comment (bxq2011hust) [inline fisco-bcos/main/main.cpp]: default is `./config.ini.start_more_group` ?
-c config.ini ==> config.ini..start_more_group ?

File: libinitializer/LedgerInitializer.h [hunk 7]
```diff
@@ -39,16 +41,17 @@ class LedgerInitializer : public std::enable_shared_from_this<LedgerInitializer>
     typedef std::shared_ptr<LedgerInitializer> Ptr;
 
     void initConfig(boost::property_tree::ptree const& _pt);
+    void startMoreLedger();
```
Comment (bxq2011hust) [inline libinitializer/LedgerInitializer.h]: what about loadNewLedger();?

File: libinitializer/LedgerInitializer.cpp [hunk 8]
```diff
@@ -92,19 +96,61 @@ void LedgerInitializer::initConfig(boost::property_tree::ptree const& _pt)
...
+        for (fs::directory_iterator iter(path); iter != endIter; iter++)
+        {
+            if (fs::extension(*iter) == ".genesis")
+            {
+                // parse group id
+                boost::property_tree::ptree pt;
+                boost::property_tree::read_ini(iter->path().string(), pt);
+                auto groupID = pt.get<int>("group.id", 0);
+
+                //
```
Comment (bxq2011hust) [inline libinitializer/LedgerInitializer.cpp]:
```suggestion

```

File: libinitializer/LedgerInitializer.cpp [hunk 9]
```diff
@@ -92,19 +96,61 @@ void LedgerInitializer::initConfig(boost::property_tree::ptree const& _pt)
                      << LOG_KV("EINFO", boost::diagnostic_information(e)) << endl;
         BOOST_THROW_EXCEPTION(e);
     }
-    /// stop the node if there is no group
-    if (m_ledgerManager->getGroupList().size() == 0)
+    return newGroupIDList;
+}
+
+vector<dev::GROUP_ID> LedgerInitializer::foreachLedgerConfigure(
+    const string& _groupConfigPath, function<bool(dev::GROUP_ID const&, const string&)> _f)
```
Comment (bxq2011hust) [inline libinitializer/LedgerInitializer.cpp]:
```suggestion
    const string& _groupConfigPath, function<bool(dev::GROUP_ID const&, const string&)> _initLedger)
```
I think we can make param more meaningful

File: fisco-bcos/main/main.cpp (hunk 6 above)
Comment (JimmyShi22) [reply fisco-bcos/main/main.cpp]: modify to ./config.ini.append_group

File: libinitializer/LedgerInitializer.cpp (hunk 8 above)
Comment (JimmyShi22) [reply libinitializer/LedgerInitializer.cpp]: Comment added

**Pull request merged**